lib/std/zig/system/x86.zig

```diff
@@ -31,13 +31,52 @@ pub fn detectNativeCpuAndFeatures(arch: Target.Cpu.Arch, os: Target.Os, cross_ta
         .features = Target.Cpu.Feature.Set.empty,
     };
 
+    const vendor, const opt_fm = detectNativeModel();
+
     // First we detect features, to use as hints when detecting CPU Model.
-    detectNativeFeatures(&cpu, os.tag);
+    detectNativeFeatures(&cpu.features, os.tag);
+
+    // Now we detect the model.
+    if (opt_fm) |fm| {
+        switch (vendor) {
+            .intel => {
+                detectIntelProcessor(&cpu, fm.family, fm.model, fm.brand_id);
+            },
+            .amd => {
+                detectAMDProcessor(&cpu, fm.family, fm.model);
+            },
+            else => {},
+        }
+    }
+
+    // Add the CPU model's feature set into the working set, but then
```

Comment:
nit: "working set" is not a good name. Can you/anyone think of a better phrase? Maybe use variable name here.

Also, maybe this should justify why this is needed and okay and does not break user assumptions on the target features: The naive reader would read that target baseline would be overwritten by current feature set of the host cpu.

Reply:
This is already existing code I moved around and I'm not exactly sure why we do this.